frontend/src/components/UserUpdateRoles.vue

```diff
@@ -344,11 +346,10 @@
             .then((rolesArray) => {
               rolesArray.forEach((clientRoles) => {
                 if (clientRoles.data.length > 0) {
-                  let lastLoginStr = LAST_LOGIN_NOT_RECORDED;
-                  if (lastLoginMap[clientRoles.clientRepresentation.name]) {
-                    lastLoginStr =
-                      lastLoginMap[clientRoles.clientRepresentation.name];
-                  }
+                  let lastLoginStr =
+                    lastLoginMap[clientRoles.clientRepresentation.clientId] ||
+                    LAST_LOGIN_NOT_RECORDED;
```

Comment:
You don't need to change it, but just for future consideration I do recommend explicitly checking `!== undefined` instead of relying on "truthy" values (undefined, null, empty string).